resources/js/components/Product/EnhancedQuantitySelect.vue

```diff
@@ -0,0 +1,54 @@
+<script>
+export default {
+    props: {
+        inputRef: {
+            type: String,
+            default: 'qty-select-1',
+        },
+        defaultQty: {
+            type: Number,
+            default: 1,
+        },
+        minQty: {
+            type: Number,
+            default: 1,
+        },
+        model: {
+            type: Number,
+            default: 1,
+        },
+        increment: {
+            type: Number,
+            default: 1,
+        },
+    },
+
+    render() {
+        return this.$scopedSlots.default(this)
+    },
+
+    methods: {
+        increase() {
+            this.updateQty(this.model + this.increment)
+        },
+
+        decrease() {
+            if (this.model - this.increment >= this.minQty) {
+                this.updateQty(this.model - this.increment)
+            }
+        },
+
+        updateQty(qty) {
+            let input = this.$scopedSlots.default()[0].context.$refs[this.inputRef]
```

Comment:
Can't we use the model?

Reply:
No, i did that before, however this won't work when in the cart. First i had a event listener on the `addtocart.vue` that would change the model qty. This won't work with the graphql-mutation component that handles it in the cart. This works for all scenarios